src/JsonSchema/TypeFactory.php

```diff
@@ -125,4 +154,43 @@ private function getClassType(?string $className, string $format = 'json', ?bool
 
         return ['$ref' => $subSchema['$ref']];
     }
+
+    /**
+     * @param array<string, mixed> $jsonSchema
+     *
+     * @return array<string, mixed>
+     *
+     * @psalm-param array{type=: string|list<string>} $jsonSchema
+     *
+     * @psalm-return array{type=: string|list<string>, $ref=: string}
+     */
+    private function addNullabilityToTypeDefinition(array $jsonSchema, Type $type) : array
+    {
+        if (! $type->isNullable()) {
+            return $jsonSchema;
+        }
+
+        if (! array_key_exists('type', $jsonSchema)) {
+            return [
```

Comment:
How would this returned literal type check the `@psalm-return array{type=: string|list<string>, $ref=: string}` if it does not have a `type` and `$ref` keys?

Reply:
Good point: will inspect if any other more precise type is possible, otherwise remove.

Reply:
899f1e3ce1c329db01c2971548b0538a44bb5ce3